src/Compilers/VisualBasic/Portable/Semantics/Conversions.vb

```diff
@@ -916,13 +916,16 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             userDefinedConversionsMightStillBeApplicable = False
             Dim conv As ConversionKind
 
-            ' Using <symbol>.IsConstant() for field accesses can result in an infinite loop.
+            ' Using source.IsConstant for field accesses can result in an infinite loop.
             ' To detect such a loop pass the already visited constants from the binder.
+            ' The check for source.IsConstant is still necessary because the node might still 
+            ' be considered as a non-constant in some error conditions (a reference before declaration,
+            ' for example).
```

Comment:
We agree offline that the code as written is good, so my suggestion would be that the comment read as follows:

```
' Using source.IsConstant for field accesses can result in an infinite loop.
' To resolve the cycle, first call GetConstantValue with the already visited constants from the binder.		             
' The check for source.IsConstant is still necessary because the node might still 
' be considered as a non-constant in some error conditions (a reference before declaration,
' for example).
```